### PR TITLE
DAO-172 Upgrade Cypress to v10

### PR DESCRIPTION
**Pull request by mcoetzee**
### What does this change?
Nothing too major, some notable changes:
- `cypress.json` -> `cypress.config.ts`
- `cypress/integration/*.spec.ts` -> `cypress/e2e/*.cy.ts`
- `plugins/index.ts` is no more

### CODE REVIEW

**Review by andreogle — APPROVED** [review]
🚀

**Review by Siegrift — APPROVED** [review]
👍 LGTM

File: cypress/support/e2e.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+//
+// You can change the location of this file or turn off
+// automatically serving support files with the
+// 'supportFile' configuration option.
+//
+// You can read more here:
+// https://on.cypress.io/configuration
+// ***********************************************************
+
+// NOTE: There is an annoying bug https://github.com/testing-library/cypress-testing-library/issues/172
```
Comment (Siegrift) [inline cypress/support/e2e.ts:16]: Btw. based on https://github.com/testing-library/cypress-testing-library/issues/172#issuecomment-1139465640 this is maybe resolved now. Can you check this when you have time?

Comment (mcoetzee) [reply cypress/support/e2e.ts:16]: I'll take a look 👍

Comment (mcoetzee) [reply cypress/support/e2e.ts:16]: No such luck. Seems to still be an issue (we've got the latest `@testing-library/cypress`)

Screenshot 2022-07-22 at 11 20 55

Comment (Siegrift) [reply cypress/support/e2e.ts:16]: 😢 Thanks for looking at this.

**Pull request merged**